oscar_odin/mappings/catalogue.py

```diff
@@ -249,3 +388,37 @@ def product_queryset_to_resources(
     return product_to_resource(
         query_set, request, user, include_children=include_children, **kwargs
     )
+
+
+def products_to_model(
+    products: List[resources.catalogue.Product], product_mapper=ProductToModel
+) -> Tuple[List[ProductModel], Dict]:
+    context = ProductModelMapperContext(ProductModel)
+
+    result = product_mapper.apply(products, context=context)
```

Comment:
This is still too weird unfortunately.
Especially the order of the statements.

So how about this:
```
result = product_mapper.apply(products, context=context)
try:
    return (list(result), context)
except TypeError:  # it is not a list
   return ([result], context)
```

Reply:
To make this more predictable and remove the need for an exception handler, the behaviour used by odin itself could be used here eg:

```python
if not hasattr(products, "__iter__"):
    products = [products]

result = product_mapper.apply(products, context=context)
return (list(result), context)
```

Or use an `isinstance` check on products against a QuerySet, but checking for __iter__ might be more reliable.